vim/autoload/himalaya/mbox.lua

```diff
@@ -3,11 +3,31 @@ local action_state = require('telescope.actions.state')
 local finders = require('telescope.finders')
 local pickers = require('telescope.pickers')
 local sorters = require('telescope.sorters')
+local previewers = require('telescope.previewers')
 
 function mbox_picker(mboxes)
   pickers.new {
     results_title = 'Mailboxes',
-    finder = finders.new_table(mboxes),
+    finder = finders.new_table {
+        results = mboxes,
+        entry_maker = function(entry)
+            return {
+                value = entry,
+                display = entry,
+                ordinal = entry,
+                preview_command = function(entry, bufnr)
+                    vim.api.nvim_buf_call(bufnr, function()
+                        local success, output = pcall(vim.fn['himalaya#mbox#post_input'], entry.value)
```

Comment:
There is an other problem. `himalaya#mbox#post_input` modifies the current mailbox and the current page: https://github.com/soywod/himalaya/blob/14d8a3fecdf34aaf0d554ed08b60e48ed019e98b/vim/autoload/himalaya/mbox.vim#L56-L58

I am afraid we will need an other function to preview without changing the cursors!

Reply:
I see! Could `curr_mbox`/`curr_page` be given as arguments to `himalaya#mbox#post_input` instead of being saved as script variables?

Reply:
We need an other function anyway, since we have to call inside the new function `himalaya#msg#list` (the tryless one from your proposition https://github.com/soywod/himalaya/pull/69#discussion_r609513786). Maybe you can rename `himalaya#msg#list` to `himalaya#msg#list_preview`, and give as parameter the `mbox`? The page can stay at 0 since it's just a preview.